File: src/nasabreakup/satellite.py
import numpy as np
from .configuration import SatType

from abc import (
    ABC,
    abstractmethod,
)


class Satellite(ABC):

    @property
    @abstractmethod
    def position(self) -> np.ndarray:
        ...

    @property
    @abstractmethod
    def velocity(self) -> np.ndarray:
        ...

    @property
    @abstractmethod
    def mass(self) -> float:
        ...

    @property
    @abstractmethod
    def characteristic_length(self) -> float:
        ...

    @property
    @abstractmethod
    def type(self) -> SatType :
        ...

File: src/satellite.rs
use crate::SatKind;
use ndarray::prelude::*;

#[derive(Debug, Clone)]
pub struct Satellite {
    pub position: Array1<f32>,
    pub velocity: Array1<f32>,
    pub mass: f32,
    pub characteristic_length: f32,
    pub sat_kind: SatKind,
}
